static/app/views/projectInstall/heartbeatFooter/useHeartbeat.tsx

```diff
@@ -0,0 +1,85 @@
+import {useState} from 'react';
+
+import {
+  Project,
+  SessionApiResponse,
+  SessionFieldWithOperation,
+  SessionStatus,
+} from 'sentry/types';
+import {useQuery} from 'sentry/utils/queryClient';
+import useOrganization from 'sentry/utils/useOrganization';
+
+const DEFAULT_POLL_INTERVAL = 5000;
+
+import partition from 'lodash/partition';
+
+type Props = {
+  project?: Project;
+};
+
+export function useHeartbeat({project}: Props) {
+  const organization = useOrganization();
+  const [firstErrorReceived, setFirstErrorReceived] = useState(false);
+  const [firstTransactionReceived, setFirstTransactionReceived] = useState(false);
+  const [sessionInProgress, setSessionInProgress] = useState(false);
+
+  const {isLoading: eventLoading} = useQuery<Project>(
+    [`/projects/${organization.slug}/${project?.slug}/`],
+    {
+      staleTime: 0,
+      refetchInterval: DEFAULT_POLL_INTERVAL,
+      enabled: !!project && !firstErrorReceived, // Fetch only if the project is available and we have not yet received an error,
+      onSuccess: data => {
+        setFirstErrorReceived(!!data.firstEvent);
+        // When an error is received, a transaction is also received
+        setFirstTransactionReceived(!!data.firstTransactionEvent);
+      },
+    }
+  );
+
+  // TODO(Priscila): Check if the query parameters are optimal
+  const {isLoading: sessionLoading} = useQuery<SessionApiResponse>(
+    [
+      `/organizations/${organization.slug}/sessions/`,
+      {
+        query: {
+          project: project?.id,
+          groupBy: 'session.status',
+          statsPeriod: '24h',
+          interval: '1h',
+          field: [SessionFieldWithOperation.SESSIONS],
```

Comment:
1. Do we need to group by session.status here? The existence of any session kind(status) proves that the SDK is connected, no? We could greatly simplify the onSuccess handler (just check the total sessions count).
2. Is there a need for an interval of 1h when we have statsPeriod 24h? I would keep both statsPeriod and interval at 1d (or even 1h).

Reply:
Thanks for the great feedback @matejminar 🌮 

1. Updated! 
2. When I use 1d/1h for both interval and statsPeriod, I don't get a fresh session. Do we need to set both? How important is it to set the interval? How does 24h differ from 1d?

Reply:
leaving only statsPeriod with 24h brings me recent sessions